contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -0,0 +1,421 @@
+    }
+    delete s_requestCommitments[requestId];
+
+    int256 weiPerUnitLink;
+    weiPerUnitLink = getFeedData();
+    if (weiPerUnitLink <= 0) {
+      revert InvalidLinkWeiPrice(weiPerUnitLink);
+    }
+    // (1e18 juels/link) * (gas/wei) / (wei/link) = juels per wei
+    uint256 juelsPerGas = (1e18 * commitment.gasPrice) / uint256(weiPerUnitLink);
+    // Gas overhead without callback
+    uint96 gasOverheadJuels = uint96(juelsPerGas * commitment.fulfillmentGas);
+    uint96 costWithoutFulfillment = gasOverheadJuels + commitment.donFee;
+
+    // The Functions Router will perform the callback to the client contract
+    IFunctionsRouter router = IFunctionsRouter(address(s_router));
+    (IFunctionsRouter.FulfillResult result, uint96 fulfillmentCostJuels) = router.fulfill(
+      requestId,
+      response,
+      err,
+      uint96(juelsPerGas),
+      costWithoutFulfillment
+    );
+
+    // Reimburse the transmitter for the fulfillment gas cost
+    s_withdrawableTokens[msg.sender] = gasOverheadJuels + fulfillmentCostJuels;
+    // Put donFee into the pool of fees, to be split later
+    // Saves on storage writes that would otherwise be charged to the user
+    s_feePool += commitment.donFee;
+
+    return result;
+  }
+
+  // ================================================================
+  // |                  Request Timeout Methods                     |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function timeoutRequest(bytes32 requestId) external override onlyRouter returns (bool) {
+    Commitment memory commitment = s_requestCommitments[requestId];
+    // Ensure that commitment exists
+    if (commitment.don == address(0)) {
+      return false;
+    }
+    // Delete commitment
+    delete s_requestCommitments[requestId];
+    emit RequestTimedOut(requestId);
+    return true;
+  }
+
+  // ================================================================
+  // |                   Node Operator methods                      |
+  // ================================================================
+  /*
+   * @notice Oracle withdraw LINK earned through fulfilling requests
+   * @notice If amount is 0 the full balance will be withdrawn
+   * @notice Both signing and transmitting wallets will have a balance to withdraw
+   * @param recipient where to send the funds
+   * @param amount amount to withdraw
+   */
+  function oracleWithdraw(address recipient, uint96 amount) external {
+    _disperseFeePool();
+
+    if (amount == 0) {
+      amount = s_withdrawableTokens[msg.sender];
+    }
+    if (s_withdrawableTokens[msg.sender] < amount) {
+      revert InsufficientBalance();
+    }
+    s_withdrawableTokens[msg.sender] -= amount;
+    IFunctionsSubscriptions router = IFunctionsSubscriptions(address(s_router));
+    router.oracleWithdraw(recipient, amount);
+  }
+
+  function _disperseFeePool() internal {
```

Comment:
This new compensation model means we must ensure _disperseFeePool() is called before we change the # of nodes in the DON.

Reply:
It is already, see the `_beforeSetConfig` hook in the Coordinator